fabric-events-interaction-v0/src/main/java/net/fabricmc/fabric/mixin/event/interaction/MixinServerPlayerInteractionManager.java

```diff
@@ -81,4 +86,30 @@ public void interactItem(ServerPlayerEntity player, World world, ItemStack stack
 			return;
 		}
 	}
+
+	@Inject(at = @At(value = "INVOKE", target = "Lnet/minecraft/block/Block;onBreak(Lnet/minecraft/world/World;Lnet/minecraft/util/math/BlockPos;Lnet/minecraft/block/BlockState;Lnet/minecraft/entity/player/PlayerEntity;)V"), method = "tryBreakBlock", locals = LocalCapture.CAPTURE_FAILHARD, cancellable = true)
+	private void breakBlock(BlockPos pos, CallbackInfoReturnable<Boolean> cir, BlockState state, BlockEntity entity, Block block) {
+		ActionResult result = BlockBreakEvents.BEFORE.invoker().beforeBlockBreak(this.world, this.player, pos, state, entity);
+
+		if (result == ActionResult.FAIL) {
+			BlockBreakEvents.CANCEL.invoker().onBlockBreakCancel(this.world, this.player, pos, state, entity);
+
+			BlockPos cornerPos = pos.add(-1, -1, -1);
+
+			for (int x = 0; x < 3; x++) {
+				for (int y = 0; y < 3; y++) {
+					for (int z = 0; z < 3; z++) {
+						this.player.networkHandler.sendPacket(new BlockUpdateS2CPacket(world, cornerPos.add(x, y, z)));
```

Comment:
Yeah, really not a fan of this, ill need to look at the vanilla code but there has to be a better way. atleast add a comment saying what it does.

Reply:
Geo, you need to still explain why the packets are sent if the event is canclled.